Source/MetroXHunter/Interaction/Locker.cpp

```diff
@@ -5,20 +5,26 @@
 #include "Interaction/Locker.h"
 #include "Interaction/InteractionComponent.h"
 #include "Interaction/InteractableComponent.h"
+#include "Interaction/BasePickUp.h"
 
+#include "Components/SceneComponent.h"
+#include "Components/WidgetComponent.h"
 #include "EnhancedInputComponent.h"
 #include "EnhancedInputSubsystems.h"
 
 ALocker::ALocker()
 {
 	PrimaryActorTick.bCanEverTick = true;
-
 	SetActorTickEnabled( false );
+
+	ItemSpawnPoint = CreateDefaultSubobject<USceneComponent>( TEXT( "SceneComponent" ) );
+	ItemSpawnPoint->SetupAttachment( RootComponent );
 }
 
 void ALocker::Interact()
 {
 	ShowSkillCheckWidget();
+	Widget->SetVisibility( false );
```

Comment:
As we talked yesterday, we should see if we can refactor that to avoid coding UI in C++